pkg/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,321 @@
+type NetworkPolicyBuilder struct {
+	// NetworkPolicy definition. Used to create networkPolicy object with minimum set of required elements.
+	Definition *netv1.NetworkPolicy
+	// Created networkPolicy object on the cluster.
+	Object *netv1.NetworkPolicy
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before NetworkPolicy object is created.
+	errorMsg string
+}
+
+// NewNetworkPolicyBuilder method creates new instance of builder.
+func NewNetworkPolicyBuilder(apiClient *clients.Settings, name, nsname string) *NetworkPolicyBuilder {
+	glog.V(100).Infof("Initializing new NetworkPolicyBuilder structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &NetworkPolicyBuilder{
+		apiClient: apiClient,
+		Definition: &netv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the networkPolicy is empty")
+
+		builder.errorMsg = "The networkPolicy 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the networkPolicy is empty")
+
+		builder.errorMsg = "The networkPolicy 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// WithNamespaceIngressRule applies ingress rule for the networkPolicy.
+func (builder *NetworkPolicyBuilder) WithNamespaceIngressRule(
+	namespaceIngressMatchLabels map[string]string,
+	podIngressMatchLabels map[string]string) *NetworkPolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Applying Ingress rule to networkPolicy %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if namespaceIngressMatchLabels == nil && podIngressMatchLabels == nil {
+		glog.V(100).Infof("At least one type of the selector for NetworkPolicy ingress rule should be defined")
+
+		builder.errorMsg = "Both namespaceIngressMatchLabels and podIngressMatchLabels parameters are empty maps"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	var peerRule netv1.NetworkPolicyPeer
+
+	if namespaceIngressMatchLabels != nil {
+		glog.V(100).Infof(
+			"Applying Ingress rule with namespaceIngressMatchLabels %v parameter to networkPolicy %s in namespace %s",
+			namespaceIngressMatchLabels, builder.Definition.Name, builder.Definition.Namespace)
+
+		peerRule.NamespaceSelector = &metav1.LabelSelector{
+			MatchLabels: namespaceIngressMatchLabels,
+		}
+	}
+
+	if podIngressMatchLabels != nil {
+		glog.V(100).Infof(
+			"Applying Ingress rule with podIngressMatchLabels %v parameter to networkPolicy %s in namespace %s",
+			podIngressMatchLabels, builder.Definition.Name, builder.Definition.Namespace)
+
+		peerRule.PodSelector = &metav1.LabelSelector{
+			MatchLabels: podIngressMatchLabels,
+		}
+	}
+
+	if builder.Definition.Spec.Ingress == nil {
```

Comment:
I think this will already be initialized. Can you check if you need it?

Reply:
as far as I remember I had problems with this

Reply:
#L133 should give you the same issue if that is the case.  Please verify if you need it. If so, align WithPolicyType with this functionality

Reply:
yea, wit was there, seems to be missed on some stage, too much changes were done since it was originally opened